woocommerce/woocommerce-ios
# Customer Search: Coordinate `Customer` and `CustomerSearchResults` entities with Storage

Part of: #7741 

### Description
This PR updates the Network's `Customer` and `WCAnalyticsCustomer` entities with a `siteID` parameter. Then adds support for upserting `Customer` and `CustomerSearchResults` to Storage.

### Changes
- Networking: Adds `siteID` to `Customer` and  `WCAnalyticsCustomer` models.
- Yosemite: Adds `upsertCustomer` and `upsertSearchCustomerResult` methods to `CustomerStore` + Unit Tests
- Storage: Adds `Customer+ReadOnlyConvertible`
- Storage: Adds necessary `StorageType+Extensions` + Unit Tests

### Testing instructions
1. Confirm that Unit Tests pass.
2. The current implementation assumes that you have at least one customer in your store, and that customer name contains the word "hello", if that's not the case, you can change this to the keyword you prefer:

https://github.com/woocommerce/woocommerce-ios/blob/ec113232b7d02b3ee9d74a56f71c7c2c7ecda6f5/WooCommerce/Classes/ViewRelated/Orders/Order%20Creation/CustomerSection/CreateOrderAddressFormViewModel.swift#L108-L112

3. Go to "Orders" > "+" > "Create new Order" > "+ "Add Customer" and tap in the "Magnifying glass" icon at the top of the screen:

<img width= 400 src="https://user-images.githubusercontent.com/3812076/192192292-26a940ff-353f-4c1b-99ba-54d443c0a91d.png">

4. You should see your siteID, the keyword used, and the number of Customers inserted to Storage, as well as the CustomerSearchResults in the console. Similar to:

```
Site ID: 121710041, keyword: hello, Customers: Optional(2)
Customer: 4,
Name: Optional("hello") Optional("merch")
Customer: 5,
Name: Optional("Hello") Optional("maldon")
```
